File: DataAccessLayer/Models/ProjectMetadata.cs
using System;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Text.Json.Serialization;

namespace DataAccessLayer.Dto
{
    public class ProjectMetadata
    {

        public ProjectMetadata()
        {
          Models = new HashSet<ModelMetadata>();
          Forms = new HashSet<FormMetadata>();
          EnumTypes = new HashSet<EnumMetadata>();

        }

        [Key]
        public int IdProjectMetadata { get; set; }
        public string Name { get; set; }
        public string Description { get; set; }
        public string Path { get; set; }
        public string DbConnectionString { get; set; }
        public string UnitOfWork { get; set; }
        public int WebApiHttpsPort { get; set; }
        public int DevServerPort { get; set; }

        
        public virtual ICollection<ModelMetadata>? Models { get; set; }
        
        public virtual ICollection<FormMetadata>? Forms { get; set; }
        
        public virtual ICollection<EnumMetadata>? EnumTypes { get; set; }
    }
}

File: ReactRedux/src/models/ProjectMetadata.ts
import { UnitOfWorkEnum, initUnitOfWorkEnum } from "../enums/UnitOfWorkEnum";
import { ModelMetadata, initModelMetadata } from "./ModelMetadata";
import { FormMetadata, initFormMetadata } from "./FormMetadata";
import { EnumMetadata, initEnumMetadata } from "./EnumMetadata";
export interface ProjectMetadata {
  idProjectMetadata:number;
  name:string;
  description?:string;
  path?:string;
  dbConnectionString?:string;
  unitOfWork:UnitOfWorkEnum;
  webApiHttpsPort?:number;
  devServerPort?:number;
  models:ModelMetadata[];
  forms:FormMetadata[];
  enumTypes:EnumMetadata[];
}

export const initProjectMetadata = {
  idProjectMetadata: 0,
  name: '',
  unitOfWork: initUnitOfWorkEnum,
  models: [],
  forms: [],
  enumTypes: [],

}
